### PR TITLE
boolean nand

### PR DESCRIPTION
**Pull request by rcraig12**
I have created this NAND Gate as the NAND gate is the most versatile of all the logic gates in that it can be used as a building block for all logic operations including the most fundamental of boolean operations AND, OR, etc.

It has been fully tested to make sure it operates correctly using a reference truth table for ensuring correctness:

A      | B      | Output
false | false | true
false | true  | true
true  | false | true
true  | true  | false

many thanks 6502Kebab AKA Robert :)

### CODE REVIEW

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        "name": "a"
+      },
+      "be8c4674-102e-4f03-a086-5ed02f3c41b6": {
+        "type": "boolean",
+        "name": "b"
+      }
+    },
+    "outputs": {
+      "a3f3a3a4-3151-4d1f-ad3d-51a4592cf219": {
+        "type": "any",
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:17]: Output type should also be `boolean`

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "boolean/nand",
+  "description": "Performs a boolean NAND operation on the values received via inputs `a` and `b`.\n\nExample:\n\n1. true@0 is received via `a`\n2. true@0 is received via `b`\n3. false@0 is sent via `nand`",
+  "interface": {
+    "inputs": {
+      "ad9e2497-ee64-46bc-ae85-86ce71a33bd9": {
+        "type": "boolean\n",
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:7]: There seems to be a trailing newline character at the end of the type definition

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "boolean/nand",
+  "description": "Performs a boolean NAND operation on the values received via inputs `a` and `b`.\n\nExample:\n\n1. true@0 is received via `a`\n2. true@0 is received via `b`\n3. false@0 is sent via `nand`",
+  "interface": {
+    "inputs": {
+      "ad9e2497-ee64-46bc-ae85-86ce71a33bd9": {
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:6]: Please add some short description to input and output ports also to help users trying to integrate the node into their project.

**Review by lvass74 — CHANGES_REQUESTED** [review]
Thanks a lot for your valuable contribution.
Suggested some minor changes so that the new prototype should match our - not yet published - standards :)

**Review by csabee1979 — COMMENTED** [review]
Agree with Laszlo. Just those some nuance changes are need. Well done!

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json (hunk 1 above)
Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json]: remedied

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json (hunk 2 above)
Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json]: remedied

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json (hunk 3 above)
Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:6]: descriptions added as suggested

**Comment by rcraig12** [comment]
hopefully the changes will be in line with your thinking :)

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 4]
```diff
@@ -4,18 +4,21 @@
   "interface": {
     "inputs": {
       "ad9e2497-ee64-46bc-ae85-86ce71a33bd9": {
-        "type": "boolean\n",
-        "name": "a"
+        "type": "boolean",
+        "name": "a",
+        "description": "Must be a boolean eg true or false"
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:9]: I'd avoid describing the type of the port in the description as the port has a type field for that.
We'd rather try to describe its purpose and if needed an example

For coherence with the other boolean nodes (like `boolean/And` I suggest this simple description:
"Receives the first operand"

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 5]
```diff
@@ -4,18 +4,21 @@
...
       },
       "be8c4674-102e-4f03-a086-5ed02f3c41b6": {
         "type": "boolean",
-        "name": "b"
+        "name": "b",
+        "description": "Must be a boolean eg true or false"
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:14]: For coherence with the other boolean nodes (like `boolean/And` I suggest this simple description:
"Receives the second operand"

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 6]
```diff
@@ -4,18 +4,21 @@
...
       }
     },
     "outputs": {
       "a3f3a3a4-3151-4d1f-ad3d-51a4592cf219": {
-        "type": "any",
-        "name": "nand"
+        "type": "boolean",
+        "name": "nand",
+        "description": "Output will either be true or false"
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:21]: For coeherence with the other boolean nodes (like `boolean/And` I suggest this simple description:
"Sends the result of the logical NAND operation."

**Review by lvass74 — CHANGES_REQUESTED** [review]
Final suggestions for better coherence with existing boolean prototypes.

**Comment by lvass74** [comment]
> hopefully the changes will be in line with your thinking :)

Almost there! 
And still grateful for your contribution! :)

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json (hunk 6 above)
Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json]: updated the description as you have suggested :)

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json (hunk 4 above)
Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json]: updated the description as you have suggested :)

**Review by lvass74 — APPROVED** [review]

**Comment by rcraig12** [comment]
CI seems to be unhappy... Looks like a dependency issue.

**Comment by lvass74** [comment]
Prototype validation was run manually as it cannot run in CI on a foreign PR. Validation was successful.

**Comment by rcraig12** [comment]
Ah ok makes sense :)

**Comment by lvass74** [comment]
@rcraig12 Sorry but would you mind to revert the last commit temporarily? Or open a new branch and a new PR for the react installer nodes?
I'd prefer to keep this PR as focused as possible and put the react installer nodes into subsequent PR.
Reverting the last commit would enable me to quickly merge this PR into our repo, then you could sync your fork and open a new PR with the reac installer nodes.
Thanks!

File: boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
+{
+  "name": "boolean/nand",
```
Comment (lvass74) [inline boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json:2]: Can you please start the name with a capital letter? like `boolean/Nand`
... again this is question of standards that we'll publish very soon now ...

Comment (rcraig12) [reply boolean/nand/a5f368aa-2cea-4110-a170-c08ec18a6a63.json]: Capitalised as requeted :)

**Comment by rcraig12** [comment]
Apologies that was my mistake not creating a new branch. Hopefully all is in order now :) React components not poduction ready as of yet in any case ;)

**Comment by lvass74** [comment]
Thank you for your contribution it is highly appreciated!

**Pull request merged**